scripts/jenkins/jenkins-job-trigger

```diff
@@ -1,96 +1,65 @@
-
-  Example config file:
-cat > jenkinsapi.cred <<EOCONF
-\$USER='myusername';
-\$APIKEY='987zyx654wvu';
-\$SCHEME='https';
-\$JURL='ci.opensuse.org';
-\$CRUMB='.crumb:123abc456def';
-EOCONF
-
-Create the Crumb string:
- curl --user <myusername>:<mypassword> 'https://ci.opensuse.org/crumbIssuer/api/xml?xpath=concat(//crumbRequestField,\":\",//crumb)'
-";
-}
-
-
-use strict;
-
-my $USER;
-my $PASSWORD; # obsolete but we keep the variable
-my $APIKEY;
-my $SCHEME="https";
-my $JURL="ci.opensuse.org";
-my $CRUMB;
-my $METHOD='POST';
-
-# read the api credentials from files
-eval `cat /etc/jenkinsapi.cred ./jenkinsapi.cred 2>/dev/null`;
-
-if ( ! $APIKEY ) {
-  print STDERR "You have not defined your APIKEY variable in the config.\n";
-  print STDERR "Login in jenkins and find your api key.";
-  print STDERR "Please add this api key to your jenkinsapi.cred file:\n";
-  print STDERR ' $APIKEY="987zyx654wvu"';
-}
-
-if ( ! $CRUMB ) {
-  print STDERR "You have not defined your CRUMB variable in the config.\n";
-  print STDERR "Please create your own crumb:\n";
-  print STDERR " curl --user <myusername>:<mypassword> 'https://ci.opensuse.org/crumbIssuer/api/xml?xpath=concat(//crumbRequestField,\":\",//crumb)'";
-  print STDERR "And then add it to your jenkinsapi.cred file:\n";
-  print STDERR ' $CRUMB=".crumb:123abc456def"';
-}
-
-$METHOD ||= 'POST';
-$SCHEME ||= 'https';
-if ($JURL =~ /river\.suse\.de/)
-{
-  $SCHEME='http';
-}
-
-my $up='';
-$up="${USER}:${APIKEY}\@" if (defined $USER && $USER ne '' && defined $APIKEY && $APIKEY ne '');
-my $jurl="$SCHEME://${up}${JURL}";
-my $job=shift || die usage();
-
-my $jobtype='normal';
-my $jtype=shift || '';
-if ($jtype =~ /^-p$/) {
-  $jobtype='parameter';
-} elsif ($jtype =~ /^-m$/) {
-  $jobtype='matrix';
-}
-
-my $triggerurl=$jurl."/job/".$job;
-my @curlargs=('--silent', '--show-error');
-push @curlargs, ("-H", "Content-Length: 0");
-
-if ($jobtype eq 'normal') {
-  $triggerurl .= "/build";
-} elsif ($jobtype eq 'parameter') {
-  $triggerurl .= "/buildWithParameters?".join('&', @ARGV);
-} elsif ($jobtype eq 'matrix') {
-  $triggerurl .= "/".join(',', @ARGV)."/build";
-}
-
-$triggerurl =~ s/ /+/g;
-push @curlargs, ("-X", $METHOD);
-push @curlargs, ("-H", $CRUMB) if ($CRUMB);
-my @cmd=("curl", @curlargs, $triggerurl);
-
-print join(' ', @cmd) if ( $ENV{debug} );
-exec(@cmd) || die $!;
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+import os
+import sys
+import json
+
+try:
+    import jenkins
+except ImportError:
+    print("Error: Dependency missing. Please install 'python-jenkins'")
```

Comment:
not sure if there is additional value in testing for that other than using import jenkins directly (it more or less provides the same error message)